### ISSUE TITLE
Mouse Events never fire on a KryptonTreeView

### ISSUE THREAD

**Issue by omarrrio** [ComponentFactory/Krypton#132]
As title says, the MouseEvents (Especially the very useful MouseDown and MouseClick) aren't firing, it seems the process ignores the events completely in favor of the NodeMouse events.

**Comment by omarrrio** [comment]
@Wagnerp yeah i just found out today when i tried doing a workaround for the `KryptonContextMenu` by making sure my mouse X & Y positions are on a node and not just anywhere in the `KryptonTreeView`, but sadly none of the Mouse Events fired up.
Thank you for looking into this.

**Comment by megakraken** [comment]
You simply need to add your event handlers to the underlying TreeView control which is exposed by KryptonTreeView.

**Comment by omarrrio** [comment]
@megakraken Hi, thanks for the reply, could you please elaborate more, because i don't understand how to add event handlers to the underlying TreeView Control, because no matter where i click, nothing fires.

**Comment by megakraken** [comment]
The _TreeView_ property of a KryptonTreeView instance gives you access to the underlying WinForms TreeView control. Just subscribe to the MouseDown event of that.

**Comment by omarrrio** [comment]
Okay so if i understand you correctly, you can fix this by adding a method to the MouseDown event, which is what i did, and it didn't fire. Oh but the MouseEnter, MouseLeave and MouseMove events fired as expected.

**Comment by Threetwosevensixseven** [comment]
Nope. Instead of subscribing to MyControl.MouseDown, subscribe to MyControl.TreeView.MouseDown.

**Comment by Wagnerp** [comment]
@megakraken & @Threetwosevensixseven 

So, do you need to initialise `protected override void OnMouseDown()`?

**Comment by Wagnerp** [comment]
Ok, so now I've added this code block:

`/// <summary>
        /// Raises the MouseDown event.
        /// </summary>
        /// <param name="e">A MouseEventArgs that contains the event data.</param>
        protected override void OnMouseDown(MouseEventArgs e)
        {
            _mouseOver = false;

            PerformNeedPaint(true);

            _treeView.Invalidate();

            base.OnMouseDown(e);
        }`

So should it work now?
